### ISSUE TITLE
Our Community Wishlist

### ISSUE THREAD

**Issue by ErinBecker** [carpentries/conversations#10]
There are lots of things that the Carpentry community is excited about developing, improving or expanding, but we don't yet have a way for community members to organize efforts around these things and make progress towards achieving them.  

There are a lot of eager individuals in the community who want to get involved, but don't know how or what to get involved with.   

Let's figure out how to bring together these ideas with this community energy and make things happen!  

A first step is gathering ideas about what people are excited about. This might be something that is already in the works, or something new. Share what you're excited about below!  

Once we've got our wishlist we can start working on connecting passion and resources to help make our wishes come true. But first let's let ourselves free range to dream. What do you wish for the Carpentry community? What do you want to see happen in the next six months or a year?  

Here are some of my wishes to get us started. Please add yours!  
- Personalized mentoring for all new Carpentry instructors  
- Ways for learners to continue participating in the Carpentry community after their workshop  
- More workshops in more regions (increase presence in Central/South America, Asia, and Africa)  
- A Data Carpentry workshop for Social Scientists  

I'd like to propose that development of this "Getting Involved" portal, including infrastructure and first-pass content be part of our second work cycle - Cycle Deimos - March 20th through May 12th. For more background about how we're using work cycles, check out our blog post: http://www.datacarpentry.org/blog/prometheus/

If you're interested in being part of the work team for developing this, please comment below.

**Comment by naupaka** [comment]
I think a carpentries twitter group that people could subscribe to would be a neat way to pull people together across geography and discipline.

**Comment by karenword** [comment]
Local meetups where participants in the same region can support each other in continued engagement.  (File under 2nd bullet point)

**Comment by dlstrong** [comment]
* More digital humanities-associated Carpentry lessons and options.
* Some kind of introduction to good database design and standards in the SQL lesson 
(I spent a chunk of time going "when you're doing it for real, don't do it this way or call it this thing, make sure to identify your units of measurement, add primary key fields, standardize naming conventions including indicators of foreign keys..." If the database as designed is meant to be an object lesson in how not to do it and things that need fixing, then well done, mission thoroughly accomplished! If it's not, well... it's serving as an object lesson anyway?)

**Comment by raynamharris** [comment]
@naupaka Have you seen this growing list of SWC and DC tweeps? Its not a group and its not exhaustive, but its a good start https://twitter.com/maneeshasane/lists/swc-dc/members

**Comment by njamescouk** [comment]
agree with @dlstrong re sql

**Comment by ErinBecker** [comment]
Thanks everyone for adding your wishes! It's really great to see what we're all excited about accomplishing. Please help me in promoting this thread so we can gather a wide variety of ideas from across our community.

**Comment by ErinBecker** [comment]
Contributed via Twitter: I also want to see courses on data analytics with Perl 6, awesome functional and OO language.

**Comment by naupaka** [comment]
+1 on how to design a database

**Comment by wrightaprilm** [comment]
Do we have a map of instructors? I was trying to plan a workshop, but couldn't find out if there were local DC instructors.

**Comment by naupaka** [comment]
How about how to use GitHub/Jekyll and markdown to make a professional website and/or share analysis results

**Comment by ErinBecker** [comment]
@naupaka Is this a suggestion for a new workshop curriculum or more like a professional development activity for instructors and/or learners?

**Comment by wrightaprilm** [comment]
Great idea @naupaka. We taught using Github Pages at University Puerto Rico-Mayaguez, but it was largely extemporaneous. Since DC doesn't do git, we did an abbreviated git session, followed by a discussion of using Github pages for communication and analysis results. I was planning to write up a similar module for the WiSE materials, but development on those has stalled out, so nothing formalized has emerged.

**Comment by karinlag** [comment]
Kind of meta, but how about a place on the websites where the community can nominate things to do, and then people can vote on it? That way we might get a better understanding of what people are missing. I've seen other sites with this feature, so I am assuming it's not rocket science to set up.

**Comment by ErinBecker** [comment]
@karinlag That's exactly the type of feature I'd love to see result from this discussion. Ideally it would incorporate a way for people to volunteer to work on specific projects so we can help facilitate working groups within the community.

**Comment by naupaka** [comment]
@ErinBecker I was thinking a lesson on GH pages that could optionally be added on to the git curriculum, and more for learners than instructors, but I can see how it might be useful for instructors too.

**Comment by mrawls** [comment]
I'd love to see a dedicated effort to bring the Carpentries to underserved populations and geographical areas in the US and Canada. Right now the workshops are highly centered in wealthy urban areas near big Universities, for obvious reasons, but I don't think this correlates well at all with the highest demand or need for the skills we teach. Some ideas could include HBCUs, Tribal Colleges/Universities, reservations, smaller towns with community colleges, etc. If there are other folks also interested in this perhaps we could band together, apply for grants, reach out to organizations, and plan some road trips.

I also really wish we could use an organization-wide Slack instead of a jumble of mailing lists and difficult-to-search GitHub Issue threads :)

**Comment by patitsas** [comment]
+1 to data carpentry for social science.

It'd be great for the social science audience to talk about keeping confidential data secure. (Same thing goes for any future data carpentry aimed at health/biomed people.)

**Comment by lexnederbragt** [comment]
A lesson on how to contribute to lessons. Maybe more general, like this impromptu workshop ["Writing and publishing on the web together using Github"](http://www.ub.uio.no/english/courses-events/courses/other/Carpentry/software-carpentry/time-and-place/170524_writing_git) I'm giving (for the second time). No lesson material (yet), I'm running it of  a script. The description is:

> You may be familiar with GitHub as a source of software, scripts, and programs. But, did you know that: 
> GitHub can also be used to work on text documents?
> 
> * GitHub enables collaboration on documents and software entirely through the web interface?
> * you can create a simple webpage with a few clicks through GitHub?
> * you don’t need any knowledge of the command-line version control tool ‘git’ to do all this?
> 
> This course will teach you how to do all these things, and more. Experience with git or GitHub or similar programs/services is not necessary, but it doesn’t hurt either.
> 
> NB time permitting, at the end the workshop, those interested can learn how to do the same operations using command line ‘git'.

I teach - using GitHub on the web only:

* setting up a repo
* commiting changes, and diffing them
* adding collaborators
* editing as collaborator directly and through pull requests
* issues, reviewing
* resolving merge conflicts
* pull request when you are not a collaborator

**Comment by lexnederbragt** [comment]
Intermediate lessons

* scientific python (numpy, scipy, ...)
* creating packages in R, python
* version control (branches, rebase)
* ...

**Comment by lexnederbragt** [comment]
Unix lesson switching to the gapminder material - so that most lessons use it

**Comment by naupaka** [comment]
@lexnederbragt yes re: bash and gapminder - this is how I've been teaching it for the last few years

**Comment by dlstrong** [comment]
+1 (+a lot) for @lexnederbragt 's suggestion of how to contribute through the web interface - I hadn't known anything about Git until 3 months after I finished instructor training, and even after having taken the Git lesson in the same session where I was teaching SQL, I'm really not confident of my ability to use "command line Git" for contributions. I much prefer the web interface.

**Comment by marwahaha** [comment]
For Carpentry in Social Science, the [D-Lab at Berkeley](https://github.com/dlab-berkeley) has done some good work. Especially, see their Python for Everything lesson: https://github.com/dlab-berkeley/python-for-everything

**Comment by gcapes** [comment]
Accurate lesson timings. This applies to most of the lessons I've taught/seen. Motivation of learners is an important aspect of the instructor training curriculum, but motivation of the trainers appears to have been overlooked. When there is too much material for the time, this can be very demotivating for the instructor because they haven't 'got through all the material'.

When there is too much material for the time, the advice has always been to go through the lesson and choose what bits you will cover/drop. The problem with this is some sections depend on previous material, so choosing what to drop is difficult and time consuming. 

A better approach would be to have the right timings for a 'core' of material, and extra material which can be slotted in if there is time.

**Comment by lexnederbragt** [comment]
@gcapes That is a good one. But the challenge is that even the same trainer can have different speeds teaching the same material depending on, for example, the group of learners. Maybe these timings should be ranges (reasonable minimums and maximums).

I really like you mentioning 'motivation of the trainers appears to have been overlooked'...

**Comment by wrightaprilm** [comment]
I think you hit the nail on the head with the variation, @lexnederbragt. In dc-py-ecology, I revised the lesson timing to how much time I have been spending when teaching the material as a 6-hour section. Revising timings was included in the release checklist, perhaps next go, maintainers could add a short paragraph on how these timings were created.
